src/mlpack/methods/preprocess/preprocess_split_main.cpp

```diff
@@ -0,0 +1,134 @@
+/**
+ * @file preprocess_split_main.cpp
+ * @author Keon Woo Kim
+ *
+ * split data CLI executable
+ */
+#include <mlpack/core.hpp>
+#include <mlpack/core/data/split_data.hpp>
+
+PROGRAM_INFO("Split into Train and Test Data", "This "
+    "utility takes data and labels and split into a training "
+    "set and a test set.");
+
+// Define parameters for data
+PARAM_STRING_REQ("input_file", "File containing data,", "i");
+// Define optional parameters
+PARAM_STRING("input_labels", "File containing labels", "I", "");
+PARAM_STRING("training_file", "File name to save train data", "t", "");
+PARAM_STRING("test_file", "File name to save test data", "T", "");
+PARAM_STRING("training_labels_file", "File name to save train label", "l", "");
+PARAM_STRING("test_labels_file", "File name to save test label", "L", "");
+
+// Define optional test ratio, default is 0.2 (Test 20% Train 80%)
+PARAM_DOUBLE("test_ratio", "Ratio of test set, if not set,"
+    "the ratio defaults to 0.2", "r", 0.2);
+
+using namespace mlpack;
+using namespace arma;
+using namespace std;
+
+int main(int argc, char** argv)
+{
+  // Parse command line options.
+  CLI::ParseCommandLine(argc, argv);
+  const string inputFile = CLI::GetParam<string>("input_file");
+  const string inputLabels = CLI::GetParam<string>("input_labels");
+  string trainingFile = CLI::GetParam<string>("training_file");
+  string testFile = CLI::GetParam<string>("test_file");
+  string trainingLabelsFile = CLI::GetParam<string>("training_labels_file");
+  string testLabelsFile = CLI::GetParam<string>("test_labels_file");
+  const double testRatio = CLI::GetParam<double>("test_ratio");
+
+  // check on data parameters
+  if (trainingFile.empty())
+  {
+    trainingFile = "train_" + inputFile;
+    Log::Warn << "You did not specify --training_file. "
+      << "Training file name is automatically set to: "
+      << trainingFile << endl;
+  }
+  if (testFile.empty())
+  {
+    testFile = "test_" + inputFile;
+    Log::Warn << "You did not specify --test_file. "
+      << "Test file name is automatically set to: " << testFile << endl;
+  }
+
+  // check on label parameters
+  if (!inputLabels.empty())
+  {
+    if (!CLI::HasParam("training_labels_file"))
+    {
+      trainingLabelsFile = "train_" + inputLabels;
+      Log::Warn << "You did not specify --training_labels_file. "
+        << "Training labels file name is automatically set to: "
+        << trainingLabelsFile << endl;
+    }
+    if (!CLI::HasParam("test_labels_file"))
+    {
+      testLabelsFile = "test_" + inputLabels;
+      Log::Warn << "You did not specify --test_labels_file. "
+        << "Test labels file name is automatically set to: "
+        << testLabelsFile << endl;
+    }
+  }
+  else
```

Comment:
We could also assume that the last dimension of the input data are the input labels, that's what the executables do.